osf/models/sanctions.py

```diff
@@ -608,6 +610,7 @@ def _on_complete(self, event_data):
             },
             auth=Auth(self.initiated_by), )
         self.save()
+        signals.after_registration_or_embargo_lifted.send(self._get_registration())
```

Comment:
Here's a fun fact: `_on_complete` is called when the Embargo has all of its required approvals, not when it's "complete". The Embargo is lifted through the same `terminate_embargo` function that the EmbargoTerminationApproval uses: https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/registrations.py#L481

Quick fix is to move this to `mark_as_completed`.

The more holistic approach might be to call this signal from `set_privacy` (and call it something like `signals.registration_made_public`?):
https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/node.py#L1206

That approach lets you remove the logic from the sanctions, and because we call `set_privacy` for each child, it eliminates the need get all of the children in the signal handler, too.

Either approach should also resolve the issue of the signal not getting sent following an accepted EmbargoTerminationApproval.

Reply:
I deleted these signals and reused the `privacy_set_public` signal with some changes, that feels like the best place to put it.